File: app/src/main/java/com/jsqix/dongqing/gank/adapter/recycler/RecyclerNormalAdapter.java
package com.jsqix.dongqing.gank.adapter.recycler;

import android.content.Context;
import android.content.Intent;
import android.view.View;

import com.jsqix.dongqing.gank.R;
import com.jsqix.dongqing.gank.WebViewActivity;
import com.jsqix.dongqing.gank.bean.DataBean;
import com.zhy.adapter.recyclerview.CommonAdapter;
import com.zhy.adapter.recyclerview.base.ViewHolder;

import java.util.List;

/**
 * Created by dongqing on 2016/12/8.
 */

public class RecyclerNormalAdapter extends CommonAdapter<DataBean> {
    public RecyclerNormalAdapter(Context context, int layoutId, List<DataBean> datas) {
        super(context, layoutId, datas);
    }

    @Override
    protected void convert(ViewHolder holder, final DataBean categorylData, int position) {
        holder.setText(R.id.text, categorylData.getDesc());
        holder.setOnClickListener(R.id.text, new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(mContext, WebViewActivity.class);
                intent.putExtra("name", categorylData.getDesc());
                intent.putExtra("url", categorylData.getUrl());
                mContext.startActivity(intent);
            }
        });
    }


}

File: app/src/main/java/com/dongqing/gank/kotlin/adapter/RecyclerNormalAdapter.kt
package com.dongqing.gank.kotlin.ui.adapter

import android.content.Context
import android.content.Intent
import com.dongqing.gank.kotlin.R
import com.dongqing.gank.kotlin.bean.gank.DataBean
import com.dongqing.gank.kotlin.ui.activity.WebViewActivity
import com.zhy.adapter.recyclerview.CommonAdapter
import com.zhy.adapter.recyclerview.base.ViewHolder

class RecyclerNormalAdapter : CommonAdapter<DataBean> {
    constructor(context: Context?, datas: List<DataBean>) : super(context, R.layout.item_recycler_normal, datas)

    override fun convert(holder: ViewHolder?, t: DataBean?, position: Int) {
        holder!!.setText(R.id.text, t!!.desc)
        holder.itemView.setOnClickListener {
            var intent = Intent(holder.itemView.context, WebViewActivity::class.java)
            intent.putExtra("name", t.desc)
            intent.putExtra("url", t.url)
            holder.itemView.context.startActivity(intent)
        }
    }
}